### PR TITLE
Feature/metadata handling

### LINKED ISSUE #10: Move notify metadata property to bugsnag
Remove this from the common message format and implement this logic on [bugsnag subscriber](https://github.com/dial-once/node-logtify-bugsnag)

### LINKED ISSUE #11: Enable conversion of metadata to json
Because winston flattens metadata [by design](https://github.com/winstonjs/winston/issues/600), we need to enable the config to JSON.stringify all the metadata on logging

Example:
we have this: 
```js
const someObject = {
  hello: 'world'
};
logger.error(`Unable to recognize `, { something: someObject });
```

This results with
```
error: [2017-08-07T16:57:12.323Z:local:ERROR:] Unable to recognize instanceId=spring,  hello=world
```

Expected:
```
error: [2017-08-07T17:03:39.600Z:local:ERROR:] Unable to recognize {
  "instanceId": "spring",
  "something": {
    "hello": "world"
  }
}
```

### LINKED ISSUE #12: Check metadata for errors and process them accordingly
If error, convert it to:
```js
{
   message: error.message,
   stack: error.stack
}
```
Or use a lib to serialize error

### PR DESCRIPTION
**Pull request by Spring3**
resolves #12 
Partially resolves #11  
Partially resolves #10

### CODE REVIEW

File: test/modules/message-spec.js [hunk 1]
```diff
@@ -78,9 +78,68 @@ describe('Message class test', () => {
       assert.equal(message.payload.meta.notify, false);
       assert.equal(message.payload.meta.something, 'else');
     });
+
+    it('should convert errors in metadata approptiately [object]', () => {
```
Comment (jkernech) [inline test/modules/message-spec.js]: ### Typo
> appropriately

File: src/modules/message.js [hunk 2]
```diff
@@ -14,25 +15,78 @@ class Message {
...
+      Object.assign(result, { error: metadata });
+    } else if (typeof metadata === 'object') {
+      Object.assign(result, metadata);
+    }
+    return result;
+  }
+
+  /**
+   * Get json interpretation of metadata
+   * @return {String} - jsonified metadata]
```
Comment (jkernech) [inline src/modules/message.js]: ### Question
Is the `]` at the end of the description intended ?

File: src/modules/message.js [hunk 3]
```diff
@@ -14,25 +15,78 @@ class Message {
...
     this.callerModuleInfo = tracer.trace();
   }
 
+  /**
+   * Convert metadata to correct form based on type
+   * @param  {any} metadata  - values/objects/arrays to be used as metadata
+   * @return {Object|string|number|boolean}        - metadata, converted to object or primitive
+   */
+  handleMetadata(metadata) {
+    // if privimite value
```
Comment (mrister) [inline src/modules/message.js]: privimite, maybe primitive ?

File: test/modules/preset-spec.js [hunk 4]
```diff
@@ -43,5 +43,12 @@ describe('Presets', () => {
     assert.equal(result.LOGSTASH_LOGGING, false);
     assert.equal(result.LOGENTRIES_LOGGING, false);
     assert.equal(result.BUGSNAG_LOGGING, false);
+    assert.equal(result.JSONIFY, true);
+  });
+
+  it('should return a piece of config object [jsonify]', () => {
+    const result = preset({ presets: ['jsonify'] });
+    assert(result);
+    assert.equal(result.JSONIFY, true);
```
Comment (mrister) [inline test/modules/preset-spec.js]: minor: mention JSONIFY in readme if not there.

Comment (Spring3) [reply test/modules/preset-spec.js]: done

**Pull request merged**